webrender/src/renderer.rs

```diff
@@ -740,10 +742,29 @@ impl Renderer {
         // Pull any pending results and return the most recent.
         while let Ok(msg) = self.result_rx.try_recv() {
             match msg {
-                ResultMsg::UpdateTextureCache(update_list) => {
+                ResultMsg::UpdateTextureData(update_list, external_image_update_list) => {
                     self.pending_texture_updates.push(update_list);
+
+                    if self.external_image_handler.is_some() {
+                        self.pending_external_image_updates.push(external_image_update_list);
+                    }
                 }
                 ResultMsg::NewFrame(frame, profile_counters) => {
+                    // When a new frame is ready, we could start to update all pending external
+                    // image request here.
+                    if !self.pending_external_image_updates.is_empty() {
```

Comment:
@glennw 
We could release the unused external image asap here. These external images will not be used because we set a new frame. If we move the deletion to the end of render() call, we just extend the life-time of these images.
What do you think for this reason?